src/main/java/core/basesyntax/service/impl/WriterImpl.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.service.Writer;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Map;
+
+public class WriterImpl implements Writer {
+    private static final String REPORT_HEAD_LINE = "fruit,quantity";
+    private static final char COMMA = ',';
+    private StorageDao storageDao;
+
+    public WriterImpl() {
+        storageDao = new StorageDaoImpl();
+    }
```

Comment:
it's not a good idea to hardcode such dependencies in the constructor. Let's pass the dao repository as a constructor parameter